iroh/src/client.rs

```diff
@@ -751,6 +794,142 @@ where
     }
 }
 
+/// Progress stream for doc import operations.
+#[derive(derive_more::Debug)]
+pub struct DocImportFileProgress {
+    #[debug(skip)]
+    stream: Pin<Box<dyn Stream<Item = Result<DocImportProgress>> + Send + Unpin + 'static>>,
+}
+
+impl DocImportFileProgress {
```

Comment:
This duplication is what lead me to the thought of exposing temp tags to the client. So that the existing impl for blobs could be reused. But I see there's other not-so-easy issues with exposing temp tags, so well...

Reply:
yea there is alot of duplication i don't know enough about all of the consequences of exposing temp tags right now, so I'm not terribly helpful here.